javascript/javascript1/week3/main.js

```diff
@@ -0,0 +1,274 @@
+addActivity("26-7-2018", "Amazon Prime", 30);
+console.log(activities);
+
+var totalUsage = 0;
+var j = 0; // extra feature to calculate with how many activities the usageLimite is reached.
+const usageLimit = 120;
+function showStatus() {
+  const totalActivities = activities.length;
+  for (var i = 0; i < activities.length; i++) {
+    totalUsage += activities[i].duration;
+    if (j === 0) {
+      if (totalUsage > usageLimit || totalUsage === usageLimit) {
+        var limitOfActivity = i + 1;
+        console.log(
+          "You have reached your usage limit with first " +
+            limitOfActivity +
+            " activities of your activities list, no more smartphoning for you!"
+        );
+        j = j + 1;
+      }
+    }
+  }
+  return [totalActivities, totalUsage];
+}
+const showMyStatus = showStatus(activities);
+console.log(
+  "You have added " +
+    showMyStatus[0] +
+    " activities. They amount to " +
+    showMyStatus[1] +
+    " min. of usage"
+);
+//FOR OPTIONAL FEATURES:
+const newActivities = [];
+var i = 0;
+var today = new Date();
+var todaysDate =
+  today.getDate() + "-" + (today.getMonth() + 1) + "-" + today.getFullYear();
+function addActivityWithoutDate(activity, duration) {
+  if (activity === undefined || duration === undefined) {
+    return "Add some activities before calling showStatus";
+  }
+  newActivities.push({});
+  for (; i < newActivities.length; i++) {
+    newActivities[i].date = todaysDate;
+    newActivities[i].activity = activity;
+    newActivities[i].duration = duration;
+  }
+}
+addActivityWithoutDate("Youtube", 30);
+addActivityWithoutDate("Facebook", 40);
+addActivityWithoutDate("News", 30);
+
+console.log(newActivities);
+
+var k = 0;
+function newShowStatus() {
+  for (i = 0; i < newActivities.length; i++) {
+    if (newActivities[i].date === todaysDate) {
+      k = k + 1;
+    }
+  }
+  return k + " number of activities on " + todaysDate;
+}
+console.log(newShowStatus());
+
+function userSpentMostTime() {
+  var array = activities;
+  var newArray = [];
+  j = 0;
+  while (array.length != 0) {
+    var anotherArray = [];
+    k = 0;
+    newArray.push({});
+    newArray[j].duration = 0;
+    for (i = 0; i < array.length; i++) {
+      if (array[i].activity === array[0].activity) {
+        newArray[j].activity = array[i].activity;
+        newArray[j].duration += array[i].duration;
+      } else {
+        anotherArray.push({});
+        anotherArray[k].activity = array[i].activity;
+        anotherArray[k].duration = array[i].duration;
+        k++;
+      }
+    }
+    j++;
+    array = anotherArray;
+  }
+  var temp_var = 0;
+  var most_used_activity = "";
+  for (x = 0; x < newArray.length; x++) {
+    if (newArray[x].duration > temp_var) {
+      temp_var = newArray[x].duration;
+      most_used_activity = newArray[x].activity;
+    } else {
+      continue;
+    }
+  }
+  return "Most used activity is: " + most_used_activity;
+}
+console.log(userSpentMostTime());
```

Comment:
I tried rewriting what you wrote. Hope it makes sense. 
I think the while-loop was to remove the date property so I made a function for that. But it isn't necessary for the next function.

```suggestion

//Removes all the date properties from an activities list
function removeDate(inputArray) {
  return inputArray.map(x => {
    return {
      activity: x.activity,
      duration: x.duration
    }
  })
}

function userSpentMostTime(activities) {
	
  const newArray = removeDate(activities);
  let temp_var = 0;

  for (let x = 0; x < newArray.length; x++) {
	//temp_var keeps track of the index of the longest activity so far
        //In the for-loop we are going through all the activites, and 
        //comparing the duration of each one to the duration of the current 
        //recorded longest activity.
    if (newArray[x].duration > Number(newArray[temp_var].duration)) {
	//When the condition is met, we save the new index of longest activity to temp_var
      temp_var = x;
    }
  }
  //we retrieve the whole object by its index and return the activity property 
  return "Most used activity is: " + newArray[temp_var].activity;
}

console.log(userSpentMostTime(activities));
```

Reply:
super!!